packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -62,23 +63,51 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
   }
 
   getTreeItem(element: TagTreeItem): vscode.TreeItem {
+    if (element.title.indexOf(TAG_SEPARATOR) > -1) {
+      return;
+    }
     return element;
   }
 
   getChildren(element?: Tag): Thenable<TagTreeItem[]> {
     if (element) {
-      const references: TagReference[] = element.notes
+      const tagChilds: TagTreeItem[] = this.tags
+        .filter(tag => tag.tag.indexOf(element.tag + TAG_SEPARATOR) > -1)
+        .map(
+          ({ tag, notes }) =>
+            new Tag(tag, tag.substring(tag.indexOf(TAG_SEPARATOR) + 1), notes)
+        );
+
+      const references: TagTreeItem[] = element.notes
         .map(({ uri }) => this.foam.workspace.get(uri))
-        .map(note => new TagReference(element.tag, note));
+        .map(note => new TagReference(element.tag, note) as TagTreeItem)
+        .concat(tagChilds);
```

Comment:
if you references separate from nested tags it will simplify your sorting function below as the first separation will already come from the two independent arrays

Reply:
I had to do all this stuff as I had issues with the promises and this was the way to get it working properly. I'll update the PR with a nicer solution I found based on your comments. Thanks!